tests/end_to_end/conftest.py

```diff
@@ -98,6 +98,20 @@ def generate_events(request, metadata):
 
         ansible_runner.run(**parameters)
 
+    yield
+    inventory_playbook = request.config.getoption('--inventory_path')
+
+    # Execute each playbook for the teardown
+    for playbook in getattr(request.module, 'teardown_playbook'):
+        teardown_playbook_path = os.path.join(getattr(request.module, 'test_data_path'), 'playbooks', playbook)
+
+        parameters = {'playbook': teardown_playbook_path, 'inventory': inventory_playbook}
+        # Check if the test case has extra variables to pass to the playbook and add them to the parameters in that case
+        if 'extra_vars' in metadata:
+            parameters.update({'extravars': metadata['extra_vars']})
+
+        ansible_runner.run(**parameters)
```

Comment:
This task should be in `clean_environment` not in `generate_events`

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/3069/commits/d48e39518cfa34dbe84fd8a2fbb254fb32006ae4